File: clients/sellingpartner-api-aa-java/src/com/amazon/SellingPartnerAPIAA/AWSAuthenticationCredentials.java
package com.amazon.SellingPartnerAPIAA;

import lombok.Builder;
import lombok.Data;
import lombok.NonNull;

/**
 * AWSAuthenticationCredentials
 */
@Data
@Builder
public class AWSAuthenticationCredentials {
    /**
     * AWS IAM User Access Key Id
     */
    @NonNull
    private String accessKeyId;

    /**
     * AWS IAM User Secret Key
     */
    @NonNull
    private String secretKey;

    /**
     * AWS Region
     */
    @NonNull
    private String region;
}

File: clients/sellingpartner-api-aa-csharp/src/Amazon.SellingPartnerAPIAA/AWSAuthenticationCredentials.cs
namespace Amazon.SellingPartnerAPIAA
{
    public class AWSAuthenticationCredentials
    {
        /**
        * AWS IAM User Access Key Id
        */
        public string AccessKeyId { get; set; }

        /**
        * AWS IAM User Secret Key
        */
        public string SecretKey { get; set; }

        /**
        * AWS Region
        */
        public string Region { get; set; }
    }
}
